### DISCUSSION TITLE
Simulating game controller programatically

### DISCUSSION THREAD

**Discussion by gitGalu** [arianrhodsandlot/nostalgist#31] [Q&A]
The question is basically about implementing touch controls for platforms that used both keyboard and game controller (joystick), mainly 8/16-bit computers.
Is it possible to programmatically simulate game controller movements without sending key presses (keyboard keys) to the emulated machine? Currently the nostalgist.pressDown(), pressUp, and press methods cause the emulated computer to register both cursor key presses and the game joystick actions (default retroarchConfig). The list of keyboard keys that can be remapped is very limited (numpad, a-z etc.). And vice versa – when we only want to use the keyboard arrow controls, the joystick actions are also triggered (at least overriding the configuration with "nul" helps here, e.g., input_player1_up: 'nul'). At the same time, when I connect a BT/USB controller, I can see that only only game controller actions are being send to the emulated machine, not keyboard key presses, which is expected behaviour. 
Is this an oversight (bug) in the current implementation of press/pressUp/pressDown, or am I doing something wrong (again)? :)

**Comment by arianrhodsandlot** [comment]
The `press` and its related methods are actually implemented by simulating keyboard input configured with `retroarchConfig`, so unfortunately we cannot send joysticks inputs without triggering keyboard inputs when corresponding inputs exist.

Since I used to mainly focus on game consoles those support joysticks only, it's enough for those simple scenarios. I have to admit simulating joystick input with keyboard input is not an ideal way to implement features like `press`, but I failed to find out a better way. (reference: https://github.com/arianrhodsandlot/nostalgist/issues/4#issuecomment-1761485219)

**Reply by gitGalu** [reply]
Thanks. Maybe it wouldn’t be that bad if  RetroArch allowed mapping using some of less common keys (for example F13-F24). 

Can I send you a private message somehow? I would like to share a project I'm working on that uses Nostalgist, without spamming here.

**Reply by arianrhodsandlot** [reply]
I'm open to developers sharing their work built with this library. Feel free to comment here or reach me out with email listed at [my GitHub profile](https://github.com/arianrhodsandlot). And if it's okay, I'm also willing to add projects built with Nostalgist.js as showcases to the documentation site!

**Reply by gitGalu** [reply]
Thanks. I just sent the email (with links, so it might end up in the spam folder :))
